### PR TITLE
Pagerduty

### PR DESCRIPTION
**Pull request by frese**
Hi - I've added a PagerDuty plugin to show incidents for the past 24 hours.
Incidents are colour coded according to status and you'r able to acknowledge / resolve incidents directly in the dropdown menu.

You will need to create a config file ".pagerduty" in your home directory, with the following content.
TOKEN: xx786567x765x76
DOMAIN: my-domain.com
USERID: XXXXXX

### CODE REVIEW

File: PagerDuty/pagerduty.30s.rb [hunk 1]
```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/ruby
+# <bitbar.title>PagerDuty</bitbar.title>
+# <bitbar.version>v1.0</bitbar.version>
+# <bitbar.a uthor>Allan Frese</bitbar.author>
```
Comment (keithamus) [inline PagerDuty/pagerduty.30s.rb]: This metadata looks a little broken. Could you fix it please?

File: PagerDuty/pagerduty.30s.rb [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+# <bitbar.author.github>frese</bitbar.author.github>
+# <bitbar.desc>Shows current pagerduty alert status.</bitbar.desc>
+# <bitbar.dependencies>ruby (httparty gem)</bitbar.dependencies>
```
Comment (keithamus) [inline PagerDuty/pagerduty.30s.rb]: It'd be awesome to see a screenshot of the plugin, linked with the `<bitbar.image>` metadata.

**Comment by keithamus** [comment]
Hey @frese great work. Could you take a look at a couple of the small points above please?

File: PagerDuty/pagerduty.30s.rb [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+    def main
+        begin
+            # Init some defaults
+            $verbose = false
+            $id      = nil
+            $command = "GET"
+            $color   = Hash.new
+            begin
+                $config  = YAML.load(File.read("#{ENV['HOME']}/.pagerduty"))
```
Comment (keithamus) [inline PagerDuty/pagerduty.30s.rb]: Convention with our other plugins has been to use environment variables for configuration. This is a bit of an unsolved problem right now - but I'm not sure plugins should be reading config from disk just yet. Could you please change these to environment variables instead?

Comment (frese) [inline PagerDuty/pagerduty.30s.rb]: I would be happy to change to environment variables - but it seems very awkward  as to how you define these in various OS X releases. ! 
And right now I can't seem to figure out how to set them in El Capitan ?? I've tried .bash_profile, /etc/launchd.conf, "launchctl setenv ..." and .MacOSX/environment.plist ?
How can I set environment variables that will be available for bitbar plugins ?

Comment (keithamus) [inline PagerDuty/pagerduty.30s.rb]: Sorry... actually I meant just standard variables at the top of the file, which consumers of the plugin just edit. Have a look at, for example, [subreedit_links.rbL6-19](https://github.com/matryer/bitbar-plugins/blob/475e51caaa0193cc0a749c768b71173fe5cc74d6/Web/Reddit/subreddit_links.rb#L6-L19) which lets you set a list of reddits to subscribe to

**Comment by frese** [comment]
So, the .pagerduty configuration file is gone :-) - now the config is in the code.

I'm looking forward that your brilliant app will be able to provide plugins with configuration data :-)

**Comment by frese** [comment]
I'm not able to find a suitable screenshot at the moment (with out it containing sensible info).
I'll add it as soon as I have something looking awesome :-)

**Comment by frese** [comment]
@keithamus - so I've added the awesome image :-)

**Comment by keithamus** [comment]
Yay! ✋

**Pull request merged**